src/core/control/jobs/PreviewJob.cpp

```diff
@@ -108,10 +108,10 @@ void PreviewJob::drawPage() {
         case RENDER_TYPE_PAGE_LAYER:
             // render single layer
             view.initDrawing(page, cr2, true);
-            if (layer == -1) {
+            if (layer == 0) {
                 view.drawBackground();
             } else {
-                Layer* drawLayer = (*page->getLayers())[layer];
+                Layer* drawLayer = (*page->getLayers())[layer - 1];
```

Comment:
index shift of one to avoid negative numbers. See consistent changes in SidebarPreviewLayerEntry